torch/csrc/distributed/c10d/CUDASymmetricMemory.cu

```diff
@@ -784,7 +784,7 @@ c10::intrusive_ptr<SymmetricMemory> CUDASymmetricMemoryAllocator::rendezvous(
   // The group_name passed to rendezvous() takes precedence over
   // the default group_name specified during allocation.
   std::string group_name_;
-  if (group_name.has_value()) {
+  if (group_name.has_value() && group_name != "") {
```

Comment:
group_name != "" alone is enough.

Reply:
Actually it is not. `std::nullopt != x` is `true` for all `x` (that are not optionals)

So the first part of the check is required to avoid UB in the line below